frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts

```diff
@@ -0,0 +1,216 @@
+import { execSync } from 'child_process';
+import { browser, ExpectedConditions as until } from 'protractor';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import {
+  click,
+  deleteResource,
+  waitForStringInElement,
+  getDropdownOptions,
+  selectDropdownOption,
+  selectDropdownOptionById,
+  createResource,
+} from '@console/shared/src/test-utils/utils';
+import {
+  consoleTypeSelector,
+  consoleTypeSelectorId,
+  rdpServiceNotConfiguredElem,
+  desktopClientTitle,
+  launchRemoteViewerButton,
+  launchRemoteDesktopButton,
+  manualConnectionTitle,
+  rdpManualConnectionTitles,
+  rdpManualConnectionValues,
+  networkSelectorId,
+} from '../views/consolesView';
+import * as dashboardView from '../views/dashboard.view';
+import * as vmView from '../views/virtualMachine.view';
+import * as disksView from '../views/vm.disks.view';
+import {
+  VM_CREATE_AND_EDIT_TIMEOUT_SECS,
+  PAGE_LOAD_TIMEOUT_SECS,
+  VM_CREATE_AND_EDIT_AND_CLOUDINIT_TIMEOUT_SECS,
+  VM_NOT_RUNNING,
+  NO_USERS_LOGGED_IN,
+  GUEST_AGENT_REQUIRED,
+  NO_ACTIVE_USERS,
+} from './utils/constants/common';
+import { VirtualMachine } from './models/virtualMachine';
+import { getWindowsVM } from './utils/templates/windowsVMForRDPL2';
+import { multusNAD } from './mocks/mocks';
+
+const VM_NAME = 'windows-rdp';
+const VM_IP = '123.123.123.123';
+
+describe('KubeVirt VM console - RDP', () => {
```

Comment:
This should be more general. Like `Tests involving guest agent`, but maybe you can think of something better